### ISSUE TITLE
Emacs fonts broken

### ISSUE THREAD

**Issue by jluttine** [NixOS/nixpkgs#82306]
**Describe the bug**

Some normal symbols (e.g., `#`, `'` and `&`) are rendered as empty rectangles in Emacs:

![image](https://user-images.githubusercontent.com/2195834/76392052-7eeee700-6379-11ea-9a6a-3414cb465857.png)

This only happens when running Emacs as its own GUI window. No problems when running Emacs inside a terminal. This happened both with my Spacemacs setting that has worked before and with vanilla Emacs.

**To Reproduce**
1. To use Vanilla emacs, just move your emacs config somewhere: `mv ~/.emacs.d ~/.emacs.d.backup`
2. Open emacs as a separate GUI window: `emacs`
3. Open some buffer and type `#`, `'` and `&`. They aren't rendered correctly.

**Expected behavior**

These symbols should be rendered correctly.

**Metadata**

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.100, NixOS, 20.03.git.a21c2fa (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - nixpkgs: `/etc/nixpkgs`
```
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: emacs
# a list of nixos modules affected by the problem
module:
```

**Comment by jluttine** [comment]
I suppose this is related to Symbola font. I removed that because it had become nonfree. See: https://www.emacswiki.org/emacs/UnicodeFonts#toc5

This is my font config in `configuration.nix`:
```
    fonts.fontconfig.defaultFonts = {
      monospace = [ "JetBrains Mono" ];
      sansSerif = [ "Noto Sans" ];
      serif = [ "Noto Serif" ];
    };
```

I have at least the following fonts installed:
```
    fonts.fonts = with pkgs; [
      jetbrains-mono
      dejavu_fonts
      noto-fonts
      hack-font
    ];
```

**Comment by jluttine** [comment]
Actually, I don't think I have removed it yet, because there were other issues that prevented me to switch to the new configuration which would have had it removed. :thinking:

**Comment by jluttine** [comment]
Hmm.. I got this fixed by upgrading my system.

For what it's worth, some possible reason why this happened:

I tried to upgrade my system to a newer version and configuration that had nonfree Symbola removed. This upgrade failed because some packages failed to build. But *maybe* the font cache was somehow mutated already? Now that I removed the packages that failed to build and completed the system upgrade, everything worked. But it seems weird if an unsuccessful switch messes the font cache. So cannot say if this is the reason. If so, it sounds a bit worrying. I'll close this anyway now that the font is working for me.
